Minesweeper.java

```diff
@@ -0,0 +1,99 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionListener;
+import java.util.*;
+import java.awt.event.*;
+
+public class Minesweeper {
+
+    // The size in pixels for the frame.
+    private static final int SIZE = 500;
+    private JFrame  frame;
+
+    private final ActionListener actionListener = actionEvent -> {
+        Object source = actionEvent.getSource();
+    };
+
+    private class Cell extends JButton {
+        private final int row;
+        private final int col;
+        private int value;
+
+        Cell(final int row, final int col,
+             final ActionListener actionListener) {
+            this.row = row;
+            this.col = col;
+            addActionListener(actionListener);
+            setText("");
+        }
+
+        //TODO: For later implementations
+        /*int getValue() {
+            return value;
+        }
+
+        void setValue(int value) {
+            this.value = value;
+        }
+
+        boolean isAMine() {
+            return value == MINE;
+        }*/
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null || getClass() != obj.getClass())
+                return false;
+            Cell cell = (Cell) obj;
+            return row == cell.row &&
+                   col == cell.col;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(row, col);
+        }
+    }
+    private class Grid {
+        private int gridSize;
+        private Cell[][] cells;
+
+        public Grid(int gridSize){
+        
+            this.gridSize = gridSize;
+            this.cells = new Cell[gridSize][gridSize];
+
+            frame = new JFrame("Minesweeper");
+            frame.setSize(SIZE, SIZE);
+            frame.setLayout(new BorderLayout());
+
+            initializeGrid();
+
+            frame.setLocationRelativeTo(null);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.setVisible(true);
+        }
+
+        private void initializeGrid() {
```

Comment:
does this need to be its own separate function? why not put it directly inside of the constructor method since you're already calling it there?

Reply:
for readability and structuring of the code, i might as well put everything together with no methods

Reply:
well the `initializeGrid` method describes (and performs) exactly what should be happening in the `Grid` constructor, which is why this seems out of place. Also, considering that we might get rid of the frame handling from here in the future, this constructor might just end up looking like 
```java
public Grid(int gridSize) {
    this.gridSize = gridSize;
    this.cells = new Cell[gridSize][gridSize];
    initializeGrid();
}
```

which is silly

Reply:
yeah but we wouldn't remove the frame handling no?

Reply:
even if we didn't, there's no point in creating a method that performs the logic for the grid constructor and then calling that method in the constructor (which is the only time it's ever going to be called), because that method is exactly what the grid constructor is supposed to do. This isn't the same as putting everything together with no methods because the grid constructor has a specific function, and doing it in the way that it is done in this commit means that there are two separate methods with the same function, which is redundant and bad practice in general

Reply:
sure you can just paste it inside then